programs/uxd/src/utils/calculate_depositories_target_redeemable_amount.rs

```diff
@@ -2,38 +2,37 @@ use anchor_lang::prelude::Result;
 use anchor_lang::require;
 
 use crate::error::UxdError;
+use crate::utils::calculate_depositories_sum_value;
 use crate::BPS_UNIT_CONVERSION;
 
 use super::checked_convert_u128_to_u64;
 use super::compute_amount_less_fraction_floor;
 
-pub struct DepositoriesTargets {
-    pub identity_depository_target_amount: u64,
-    pub mercurial_vault_depository_0_target_amount: u64,
-    pub credix_lp_depository_0_target_amount: u64,
+pub struct DepositoriesTargetRedeemableAmount {
+    pub identity_depository_target_redeemable_amount: u64,
+    pub mercurial_vault_depository_0_target_redeemable_amount: u64,
+    pub credix_lp_depository_0_target_redeemable_amount: u64,
 }
 
-pub fn calculate_depositories_targets(
+pub fn calculate_depositories_target_redeemable_amount(
```

Comment:
Same here, we could create a vec to simplify the utils function:

```
pub struct CalculateDepositoriesTargetRedeemableAmountParams {
    weight_bps: u16,
    redeemable_circulating_supply: u128,
    redeemable_amount_under_management_cap: u128,
}

pub fn calculate_depositories_target_redeemable_amount(
    params: Vec<CalculateDepositoriesTargetRedeemableAmountParams>
) -> Result<DepositoriesTargetRedeemableAmount> {
```

Reply:
Actually I wanted to avoid using a vector and made this a utility so that everywhere in the code we know for sure we are taking into account ALL depositories related to rebalancing.

for example if we add a new depository, we just need to update that function's signature and then we get a compilation error everywhere in the code where we need to pass a value

Reply:
Also not using a vector saves on computation since there is no memory allocation involved in just passing parameters from the stack

Reply:
For example here:
https://github.com/UXDProtocol/uxd-program/pull/252/files#diff-9153007ddc0f7276312abc33a288a4535786a530dcc0543818b49d34a18fc988R64

It forces the writer of the code to consider what to do with all depositories, which helps understand the algorithm.

Reply:
> Actually I wanted to avoid using a vector and made this a utility so that everywhere in the code we know for sure we are taking into account ALL depositories related to rebalancing.
>
> for example if we add a new depository, we just need to update that function's signature and then we get a compilation error everywhere in the code where we need to pass a value.

If I'm adding support for one depository, I would be happy to not have to modify and understand 5 files/funtions. Utility functions are functions you should not have to modify, they are here to be used by bigger-scoped functions to do a simple task.

> Also not using a vector saves on computation since there is no memory allocation involved in just passing parameters from the stack, and since this function is used everywhere, this adds up a lot on the complicated maths IXs

Not sure computation is an issue for the kind of program we are running. Do we run into computation issue as of now?

Reply:
Thinking the same as Greg here, gonna be hard to add depositories later and think or what to adapt and what not.

ideally should find a design here when you add a new depository, it errors or fail if not added where it should (either using a check with a total depository count and a vector or some other strat)

Reply:
> If I'm adding support for one depository, I would be happy to not have to modify and understand 5 files/funtions. Utility functions are functions you should not have to modify, they are here to be used by bigger-scoped functions to do a simple task.

True, but in this case in my mind: `the annoyance is a feature not a bug` this is because if you dont consider the files you may silently introduce a bug, which could be very difficult to understand/detect

Reply:
> Thinking the same as Greg here, gonna be hard to add depositories later and think or what to adapt and what not.
> 
> ideally should find a design here when you add a new depository, it errors or fail if not added where it should (either using a check with a total depository count and a vector or some other strat)

Yeah I definitly understand its an overhead, but I think the overhead is trading code development time vs safety:
 - safer because you have to adjust every callsite to check that you did add your new depository
 - less error prone because:
     - the order of parameter is enforced by the naming of the parameters (unlike in a vector where ordering is not guaranteed)
     - the number of parameter can be detected at compile time rather than at run time

Reply:
I think trading:
 - adding a depository quickly is more difficult
 - we are more resilient to typos and tiny errors that are difficult to detect
 
 I think is a good tradeoff to take imo

Reply:
I feel you, no worries. I think debates are good overall, but text on PR definitly is not the best format for it.

Maybe in the future we can schedule a 30 minute slot where we can code review by voice and have quick chats to clear up this sort of issues

Reply:
> if one day we add a depository, nothing will remind us of where to modify (no error will appear right? You commented it’s a feature but I don’t think it will warn you? I read on mobile so maybe I miss smthg)

Ah I think i didnt understand this part, yeah nothing is warning us when adding a new depository. But you will be warned if you add the depository to the super-mint and super-redeem, and then it will warn you everywhere in the code further if you do. that's what I meant the feature is